src/SFA.DAS.EmployerAccounts.ReadStore/SFA.DAS.EmployerAccounts.ReadStore/Models/AccountUser.cs

```diff
@@ -141,7 +142,7 @@ private void EnsureUserHasBeenRemoved()
         {
             if (Removed == null)
             {
-                throw new InvalidOperationException("User has not been removed");
+                throw new UserNotRemovedException("User has not been removed");
```

Comment:
I think the issue raised with the ticket is because of the EnsureUserHasBeenRemoved method.

If my understanding is correct...

The process is creating a user that either does not exist or has been removed?

If the user already exists then we try to re-create the user but only if the recreate date is greater than the created date.

The first check is to check that the user has actually been previously removed.  
Looking at the user.Remove method if a user has been removed then the removed property is set to the current date.

Therefore when we are checking that the user to be recreated has actually been removed then we would expect the Removed property to have a value rather than null.

So the EnsureUserHasBeenRemoved is incorrect.  It should be 

if (Removed.HasValue)
{
       throw new UserNotRemovedException("User has not been removed");
}

That is the only change required in this ticket.  

Please correct me if my understanding is wrong.